Source/Core/Core/PrimeHack/Mods/FpsControls.cpp

```diff
@@ -263,72 +249,69 @@ void FpsControls::run_mod_mp2(Region region) {
+    if (HandleReticleLockOn()) {
+      handle_cursor(cursor_base + 0x9c, cursor_base + 0x15c, 0.95f, 0.90f);
+    }
+    else if (beamvisor_menu) { // If we unlock the reticle anyway, no reason to perform this.
+      u32 mode = read32(read32(mp2_static.beamvisor_menu_address) + mp2_static.beamvisor_menu_offset + 0xc);
 
-      if (HandleReticleLockOn()) {
+      // if the menu id is not null
+      if (mode != 0xFFFFFFFF) {
         handle_cursor(cursor_base + 0x9c, cursor_base + 0x15c, 0.95f, 0.90f);
       }
-      else if (beamvisor_menu) { // If we unlock the reticle anyway, no reason to perform this.
-        u32 mode = read32(read32(mp2_static.beamvisor_menu_address) + 0x1bc + 0x190);
-
-        // if the menu id is not null
-        if (mode != 0xFFFFFFFF) {
-          handle_cursor(cursor_base + 0x9c, cursor_base + 0x15c, 0.95f, 0.90f);
-        }
-      }
     }
+  } else {
+    // the cursor won't center itself on NTSC-J when going into the beam/visor menu :S
+    // for safety applying it on other regions of the game
+    writef32(0.0f, cursor_base + 0x9c);
+    writef32(0.0f, cursor_base + 0x15c);
 
-    return;
-  }
-  else {
     calculate_pitch_delta();
-  }
-
-
-  // Grab the arm cannon address, go to its transform field (NOT the
-  // Actor's xf @ 0x30!!)
-  u32 arm_cannon_model_matrix = read32(cplayer_address + 0xea8) + 0x3b0;
-  writef32(pitch, cplayer_address + 0x5f0);
-  // For whatever god forsaken reason, writing pitch to the z component of the
-  // right vector for this xf makes the gun not lag. Won't fix what ain't broken
-  writef32(pitch, arm_cannon_model_matrix + 0x24);
-  u32 tweak_player_address = 0;
-  if (region == Region::NTSC_U) {
-    tweak_player_address = read32(read32(GPR(13) - 0x6410));
-  }
-  else if (region == Region::NTSC_J) {
-    tweak_player_address = read32(read32(GPR(13) - 0x63f8));
-  }
-  else if (region == Region::PAL) {
-    tweak_player_address = read32(read32(GPR(13) - 0x6368));
-  }
-  if (mem_check(tweak_player_address)) {
-    // This one's stored as degrees instead of radians
-    writef32(87.0896f, tweak_player_address + 0x180);
-  }
+    // Grab the arm cannon address, go to its transform field (NOT the
+    // Actor's xf @ 0x30!!)
+    u32 arm_cannon_model_matrix = read32(cplayer_address + 0xea8) + 0x3b0;
+    writef32(pitch, cplayer_address + 0x5f0);
+    // For whatever god forsaken reason, writing pitch to the z component of the
+    // right vector for this xf makes the gun not lag. Won't fix what ain't broken
+    writef32(pitch, arm_cannon_model_matrix + 0x24);
+    u32 tweak_player_address = 0;
+    if (region == Region::NTSC_U) {
+      tweak_player_address = read32(read32(GPR(13) - 0x6410));
+    }
+    else if (region == Region::NTSC_J) {
+      tweak_player_address = read32(read32(GPR(13) - 0x63f8));
+    }
+    else if (region == Region::PAL) {
+      tweak_player_address = read32(read32(GPR(13) - 0x6368));
+    }
```

Comment:
This would be better if you made a mp2_static for the tweak offset since that's the only thing that changes between versions.